Un-annotated tag, such as from a release in GitHub, causes task to fail.

Recreation:

* Use GitHub
* Push a few changes to `master`
* Create a release `v.0.0.1` which creates a tag at that commit within `master`
* Observe that the tag appears in my local repository when pulled
* Azure DevOps release kicks off for that tag, `v0.0.1`

Expected behavior: The version numbers are populated correctly

Actual behavior: The below output --

```
2020-10-12T01:03:14.6130427Z ##[section]Starting: ExtractVersionFromTag
2020-10-12T01:03:14.6251807Z ==============================================================================
2020-10-12T01:03:14.6252394Z Task         : Extract Version
2020-10-12T01:03:14.6252779Z Description  : Extract the version from the git tag.
2020-10-12T01:03:14.6253236Z Version      : 1.1.0
2020-10-12T01:03:14.6253573Z Author       : Damien Aicheh
2020-10-12T01:03:14.6253898Z Help         : 
2020-10-12T01:03:14.6254270Z ==============================================================================
2020-10-12T01:03:14.8663633Z [command]C:\agents\2.175.2\externals\git\cmd\git.exe describe --abbrev=0
2020-10-12T01:03:14.9007219Z fatal: No annotated tags can describe 'dfe4fbaf0a25ef01846f2dbc5c071e62dc0446a7'.
2020-10-12T01:03:14.9008539Z However, there were unannotated tags: try --tags.
2020-10-12T01:03:14.9043878Z ##[error]No tag found on this branch, please verify you have one in your remote repository.
2020-10-12T01:03:14.9226511Z ##[error]Exit code 1 returned from process: file name 'C:\agents\2.175.2\externals\node\bin\node.exe', arguments '"D:\a\_tasks\ExtractVersionFromTag_e878d427-f746-4ef5-ae63-c85002ded617\1.1.0\extract-version-from-tag.js"'.
2020-10-12T01:03:14.9233595Z ##[section]Finishing: ExtractVersionFromTag
```

I believe that adding `--tags` to the args as described in the command output would indeed fix this, so I'll create a PR to attempt that change.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,0 +1,14 @@
+# Change Log
+
+### 1.0.5
+
+Update extract version from with `--tags` option.
+Add more logs details if an error occured.
+
+### 1.0.4
+
+Manage pre-release tags.
+
+### 1.0.0
+
+Stable version.

--- a/tasks/ExtractVersionFromTagTask/extract-version-from-tag.js
+++ b/tasks/ExtractVersionFromTagTask/extract-version-from-tag.js
@@ -56,7 +56,7 @@ function run() {
                 task.debug("Moving to " + projectFolderPath);
                 task.cd(projectFolderPath);
                 git = task.which('git', true);
-                args = ["describe", "--abbrev=0"];
+                args = ["describe", "--tags", "--abbrev=0"];
                 tagResult = task.execSync(git, args);
                 if (tagResult.code !== 0) {
                     if (tagResult.error != null) {

--- a/tasks/ExtractVersionFromTagTask/extract-version-from-tag.js
+++ b/tasks/ExtractVersionFromTagTask/extract-version-from-tag.js
@@ -8,111 +8,79 @@ var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, ge
         step((generator = generator.apply(thisArg, _arguments || [])).next());
     });
 };
-var __generator = (this && this.__generator) || function (thisArg, body) {
-    var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
-    return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
-    function verb(n) { return function (v) { return step([n, v]); }; }
-    function step(op) {
-        if (f) throw new TypeError("Generator is already executing.");
-        while (_) try {
-            if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
-            if (y = 0, t) op = [op[0] & 2, t.value];
-            switch (op[0]) {
-                case 0: case 1: t = op; break;
-                case 4: _.label++; return { value: op[1], done: false };
-                case 5: _.label++; y = op[1]; op = [0]; continue;
-                case 7: op = _.ops.pop(); _.trys.pop(); continue;
-                default:
-                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
-                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
-                    if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
-                    if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
-                    if (t[2]) _.ops.pop();
-                    _.trys.pop(); continue;
-            }
-            op = body.call(thisArg, _);
-        } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
-        if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
-    }
-};
-exports.__esModule = true;
-var task = require("azure-pipelines-task-lib/task");
-var fs = require("fs");
-var MAJOR = 'MAJOR';
-var MINOR = 'MINOR';
-var PATCH = 'PATCH';
-var PRE_RELEASE = 'PRE_RELEASE';
-var NUMBER_OF_COMMITS = 'NUMBER_OF_COMMITS';
+Object.defineProperty(exports, "__esModule", { value: true });
+const task = require("azure-pipelines-task-lib/task");
+const fs = require("fs");
+const MAJOR = 'MAJOR';
+const MINOR = 'MINOR';
+const PATCH = 'PATCH';
+const PRE_RELEASE = 'PRE_RELEASE';
+const NUMBER_OF_COMMITS = 'NUMBER_OF_COMMITS';
 function run() {
-    return __awaiter(this, void 0, void 0, function () {
-        var projectFolderPath, git, args, tagResult, tag, tagSplitted, versionsIndicator, preSplit, args, result, numberOfCommits;
-        return __generator(this, function (_a) {
-            try {
-                projectFolderPath = task.getPathInput('projectFolderPath');
-                if (!fs.existsSync(projectFolderPath)) {
-                    task.error("Source directory does not exist: " + projectFolderPath);
-                    process.exit(1);
-                }
-                task.debug("Moving to " + projectFolderPath);
-                task.cd(projectFolderPath);
-                git = task.which('git', true);
-                args = ["describe", "--tags", "--abbrev=0"];
-                tagResult = task.execSync(git, args);
-                if (tagResult.code !== 0) {
-                    if (tagResult.error != null) {
-                        task.error(tagResult.error.name + " " + tagResult.error.message);
-                        task.error(tagResult.error.stack);
-                    }
-                    task.error('No tag found on this branch, please verify you have one in your remote repository.');
-                    process.exit(1);
-                }
-                task.debug("Tag retreived: " + tagResult.stdout);
-                tag = tagResult.stdout;
-                if (tag.includes('v')) {
-                    tagSplitted = tag.split('v');
-                    tag = tagSplitted[1];
-                }
-                if (tag.includes('\n')) {
-                    tag = tag.split('\n')[0];
-                }
-                versionsIndicator = tag.split('.');
-                task.debug(versionsIndicator.toString());
-                if (versionsIndicator[2].includes('-')) {
-                    preSplit = versionsIndicator[2].split('-');
-                    // Replacing PATCH split with pre release tag split
-                    versionsIndicator[2] = preSplit[0];
-                    setVariableOrDefault(PRE_RELEASE, preSplit[1]);
-                }
-                else {
-                    // setting empty string a PRE_RELEASE
-                    task.setVariable(PRE_RELEASE, '');
-                }
-                setVariableOrDefault(MAJOR, versionsIndicator[0]);
-                setVariableOrDefault(MINOR, versionsIndicator[1]);
-                setVariableOrDefault(PATCH, versionsIndicator[2]);
-                task.debug('Get the number of commit until this tag');
-                args = ["rev-list", "--count", "HEAD"];
-                result = task.execSync(git, args);
-                numberOfCommits = result.stdout.split('\n');
-                setVariableOrDefault(NUMBER_OF_COMMITS, numberOfCommits[0]);
-                task.debug("Major:" + task.getVariable(MAJOR));
-                task.debug("Minor:" + task.getVariable(MINOR));
-                task.debug("Patch:" + task.getVariable(PATCH));
-                if (task.getVariable(PRE_RELEASE)) {
-                    task.debug("Pre Release:" + task.getVariable(PRE_RELEASE));
-                }
-                task.debug("Number of commits:" + task.getVariable(NUMBER_OF_COMMITS));
-                task.setResult(task.TaskResult.Succeeded, "Extract version from tag succeeded");
+    return __awaiter(this, void 0, void 0, function* () {
+        try {
+            var projectFolderPath = task.getPathInput('projectFolderPath');
+            if (!fs.existsSync(projectFolderPath)) {
+                task.error(`Source directory does not exist: ${projectFolderPath}`);
+                process.exit(1);
             }
-            catch (err) {
-                task.setResult(task.TaskResult.Failed, err.message);
+            task.debug(`Moving to ${projectFolderPath}`);
+            task.cd(projectFolderPath);
+            let git = task.which('git', true);
+            var args = ["describe", "--tags", "--abbrev=0"];
+            let tagResult = task.execSync(git, args);
+            if (tagResult.code !== 0) {
+                if (tagResult.error != null) {
+                    task.error(`${tagResult.error.name} ${tagResult.error.message}`);
+                    task.error(tagResult.error.stack);
+                }
+                task.error('No tag found on this branch, please verify you have one in your remote repository.');
+                process.exit(1);
             }
-            return [2 /*return*/];
-        });
+            task.debug(`Tag retreived: ${tagResult.stdout}`);
+            var tag = tagResult.stdout;
+            if (tag.includes('v')) {
+                var tagSplitted = tag.split('v');
+                tag = tagSplitted[1];
+            }
+            if (tag.includes('\n')) {
+                tag = tag.split('\n')[0];
+            }
+            var versionsIndicator = tag.split('.');
+            task.debug(versionsIndicator.toString());
+            if (versionsIndicator[2].includes('-')) {
+                const preSplit = versionsIndicator[2].split('-');
+                // Replacing PATCH split with pre release tag split
+                versionsIndicator[2] = preSplit[0];
+                setVariableOrDefault(PRE_RELEASE, preSplit[1]);
+            }
+            else {
+                // setting empty string a PRE_RELEASE
+                task.setVariable(PRE_RELEASE, '');
+            }
+            setVariableOrDefault(MAJOR, versionsIndicator[0]);
+            setVariableOrDefault(MINOR, versionsIndicator[1]);
+            setVariableOrDefault(PATCH, versionsIndicator[2]);
+            task.debug('Get the number of commit until this tag');
+            var args = ["rev-list", "--count", "HEAD"];
+            let result = task.execSync(git, args);
+            var numberOfCommits = result.stdout.split('\n');
+            setVariableOrDefault(NUMBER_OF_COMMITS, numberOfCommits[0]);
+            task.debug(`Major:` + task.getVariable(MAJOR));
+            task.debug(`Minor:` + task.getVariable(MINOR));
+            task.debug(`Patch:` + task.getVariable(PATCH));
+            if (task.getVariable(PRE_RELEASE)) {
+                task.debug(`Pre Release:` + task.getVariable(PRE_RELEASE));
+            }
+            task.debug(`Number of commits:` + task.getVariable(NUMBER_OF_COMMITS));
+            task.setResult(task.TaskResult.Succeeded, "Extract version from tag succeeded");
+        }
+        catch (err) {
+            task.setResult(task.TaskResult.Failed, err.message);
+        }
     });
 }
-function setVariableOrDefault(name, value, defaultValue) {
-    if (defaultValue === void 0) { defaultValue = '0'; }
+function setVariableOrDefault(name, value, defaultValue = '0') {
     if (value) {
         task.setVariable(name, value);
     }

--- a/tasks/ExtractVersionFromTagTask/extract-version-from-tag.ts
+++ b/tasks/ExtractVersionFromTagTask/extract-version-from-tag.ts
@@ -20,7 +20,7 @@ async function run() {
         task.cd(projectFolderPath);
 
         let git: string = task.which('git', true);
-        var args = ["describe", "--abbrev=0"];
+        var args = ["describe", "--tags", "--abbrev=0"];
 
         let tagResult = task.execSync(git, args);
 

--- a/tasks/ExtractVersionFromTagTask/task.json
+++ b/tasks/ExtractVersionFromTagTask/task.json
@@ -13,7 +13,7 @@
     "author": "Damien Aicheh",
     "version": {
         "Major": 1,
-        "Minor": 1,
+        "Minor": 2,
         "Patch": 0
     },
     "inputs": [

--- a/tasks/UpdateAndroidVersionGradleTask/task.json
+++ b/tasks/UpdateAndroidVersionGradleTask/task.json
@@ -14,7 +14,7 @@
     "version": {
         "Major": 1,
         "Minor": 0,
-        "Patch": 0
+        "Patch": 1
     },
     "inputs": [
         {

--- a/tasks/UpdateAndroidVersionGradleTask/update-android-version-gradle.js
+++ b/tasks/UpdateAndroidVersionGradleTask/update-android-version-gradle.js
@@ -16,7 +16,7 @@ function run() {
         try {
             var buildGradlePath = task.getPathInput('buildGradlePath');
             if (!fs.existsSync(buildGradlePath)) {
-                task.error(`The file path does not exist: ${buildGradlePath}`);
+                task.error(`The file path for the build.gradle does not exist or is not found: ${buildGradlePath}`);
                 process.exit(1);
             }
             task.debug(`Running task with ${buildGradlePath}`);

--- a/tasks/UpdateAndroidVersionGradleTask/update-android-version-gradle.ts
+++ b/tasks/UpdateAndroidVersionGradleTask/update-android-version-gradle.ts
@@ -6,7 +6,7 @@ async function run() {
         var buildGradlePath = task.getPathInput('buildGradlePath');
 
         if (!fs.existsSync(buildGradlePath)) {
-            task.error(`The file path does not exist: ${buildGradlePath}`);
+            task.error(`The file path for the build.gradle does not exist or is not found: ${buildGradlePath}`);
             process.exit(1);
         }
 

--- a/tasks/UpdateAndroidVersionManifestTask/task.json
+++ b/tasks/UpdateAndroidVersionManifestTask/task.json
@@ -14,7 +14,7 @@
     "version": {
         "Major": 1,
         "Minor": 0,
-        "Patch": 0
+        "Patch": 1
     },
     "inputs": [
         {

--- a/tasks/UpdateAndroidVersionManifestTask/update-android-version-manifest.js
+++ b/tasks/UpdateAndroidVersionManifestTask/update-android-version-manifest.js
@@ -16,7 +16,7 @@ function run() {
         try {
             var androidManifestPath = task.getPathInput('androidManifestPath');
             if (!fs.existsSync(androidManifestPath)) {
-                task.error(`The file path does not exist:: ${androidManifestPath}`);
+                task.error(`The file path for the AndroidManifest.xml does not exist or is not found: ${androidManifestPath}`);
                 process.exit(1);
             }
             task.debug(`Running task with ${androidManifestPath}`);

--- a/tasks/UpdateAndroidVersionManifestTask/update-android-version-manifest.ts
+++ b/tasks/UpdateAndroidVersionManifestTask/update-android-version-manifest.ts
@@ -6,7 +6,7 @@ async function run() {
         var androidManifestPath = task.getPathInput('androidManifestPath');
 
         if (!fs.existsSync(androidManifestPath)) {
-            task.error(`The file path does not exist:: ${androidManifestPath}`);
+            task.error(`The file path for the AndroidManifest.xml does not exist or is not found: ${androidManifestPath}`);
             process.exit(1);
         }
 

--- a/tasks/UpdateiOSVersionInfoPlistTask/task.json
+++ b/tasks/UpdateiOSVersionInfoPlistTask/task.json
@@ -14,7 +14,7 @@
     "version": {
         "Major": 1,
         "Minor": 0,
-        "Patch": 0
+        "Patch": 1
     },
     "inputs": [
         {

--- a/tasks/UpdateiOSVersionInfoPlistTask/update-ios-version-info-plist.js
+++ b/tasks/UpdateiOSVersionInfoPlistTask/update-ios-version-info-plist.js
@@ -17,7 +17,7 @@ function run() {
         try {
             var infoPlistPath = task.getPathInput('infoPlistPath');
             if (!fs.existsSync(infoPlistPath)) {
-                task.error(`The file path does not exist: ${infoPlistPath}`);
+                task.error(`The file path for the info.plist does not exist or is not found: ${infoPlistPath}`);
                 process.exit(1);
             }
             task.debug(`Running task with ${infoPlistPath}`);

--- a/tasks/UpdateiOSVersionInfoPlistTask/update-ios-version-info-plist.ts
+++ b/tasks/UpdateiOSVersionInfoPlistTask/update-ios-version-info-plist.ts
@@ -7,7 +7,7 @@ async function run() {
         var infoPlistPath = task.getPathInput('infoPlistPath');
 
         if (!fs.existsSync(infoPlistPath)) {
-            task.error(`The file path does not exist: ${infoPlistPath}`);
+            task.error(`The file path for the info.plist does not exist or is not found: ${infoPlistPath}`);
             process.exit(1);
         }
 

--- a/vss-extension.json
+++ b/vss-extension.json
@@ -2,7 +2,7 @@
     "manifestVersion": 1,
     "id": "mobile-versioning-task",
     "name": "Mobile Versioning",
-    "version": "1.0.4",
+    "version": "1.0.5",
     "publisher": "damienaicheh",
     "public": true,
     "targets": [

